scanpopup.cc

```diff
@@ -467,28 +482,49 @@ Qt::WindowFlags ScanPopup::unpinnedWindowFlags() const
 
 void ScanPopup::translateWordFromClipboard()
 {
-	return translateWordFromClipboard(QClipboard::Clipboard);
+  TranslateWordFromClipboardMode = QClipboard::Clipboard;
+  setTranslateWordFromClipboard();
 }
 
 void ScanPopup::translateWordFromSelection()
 {
-	return translateWordFromClipboard(QClipboard::Selection);
+  TranslateWordFromClipboardMode = QClipboard::Selection;
+  setTranslateWordFromClipboard();
+}
+
+void ScanPopup::setTranslateWordFromClipboard()
+{
+  isTranslateWordFromClipboard = true;
+  translateWordFromClipboardDelayTimer.start();
+  translateWordFromClipboardClearTimer.start();
 }
 
 void ScanPopup::editGroupRequested()
 {
   emit editGroupRequested( ui.groupList->getCurrentGroup() );
 }
 
-void ScanPopup::translateWordFromClipboard(QClipboard::Mode m)
+void ScanPopup::handleTranslateWordFromClipboard()
 {
+  if(!isTranslateWordFromClipboard) return;
```

Comment:
Code style:
* move `return;` to the next line;
* pad with spaces inside parentheses.

Reply:
fixed